packages/web-components/fast-element/src/styles/css-binding-directive.ts

```diff
@@ -0,0 +1,67 @@
+import type { Binding, BindingSource } from "../binding/binding.js";
+import type { Subscriber } from "../observation/notifier.js";
+import type { ExpressionObserver } from "../observation/observable.js";
+import { AddBehavior, CSSDirective } from "./css-directive.js";
+import type { ComposableStyles } from "./element-styles.js";
+import type { HostBehavior, HostController } from "./host.js";
+
+/**
+ * Enables bindings in CSS.
+ *
+ * @public
+ */
+export class CSSBindingDirective
+    implements HostBehavior, Subscriber, CSSDirective, BindingSource {
+    /**
+     * Creates an instance of CSSBindingDirective.
+     * @param dataBinding - The binding to use in CSS.
+     * @param targetAspect - The CSS property to target.
+     */
+    public constructor(
+        public readonly dataBinding: Binding,
+        public readonly targetAspect: string
+    ) {}
+
+    /**
+     * Creates a CSS fragment to interpolate into the CSS document.
+     * @returns - the string to interpolate into CSS
+     */
+    createCSS(add: AddBehavior): ComposableStyles {
+        add(this);
+        return `var(${this.targetAspect})`;
+    }
+
+    /**
+     * Executed when this behavior's host is connected.
+     * @param controller - Controls the behavior lifecycle.
+     */
+    connectedCallback(controller: HostController): void {
+        const observer: ExpressionObserver =
+            (controller as any)[this.targetAspect] ??
+            ((controller as any)[this.targetAspect] = this.dataBinding.createObserver(
+                this,
+                this
+            ));
+
+        (observer as any).controller = controller;
+
+        this.handleChange(null, observer);
+    }
+
+    /**
+     * Called when a subject this instance has subscribed to changes.
+     * @param subject - The subject of the change.
+     * @param args - The event args detailing the change that occurred.
+     *
+     * @internal
+     */
+    handleChange(_: any, observer: ExpressionObserver): void {
+        const controller = (observer as any).controller;
+        (controller.source as HTMLElement).style.setProperty(
```

Comment:
One thought I have here is that setting the style attribute for the element will remove any properties set by the binding, and that this will effectively "hoist" a style attribute to the host. It's probably an edge case, but I'm sure *someone* will hit it eventually. Did you consider using a constructable stylesheet instead? It would surely be less performant but also more durable. Maybe we can enable a strategy in a follow-up for CSSBindingDirectives to adjust the property-setting behavior?

Reply:
Hmm. I didn't think about that. Creating an additional stylesheet might make this cost prohibitive, especially since it's likely just to enable one or two properties and it would mean creating a sheet per component instance, not per type.

Reply:
I've got a fix coming for this.